### ISSUE TITLE
check_ping command not working on Alpine

### ISSUE THREAD

**Issue by BE-Digital-Solutions-Bart** [manios/docker-nagios#93]
I'm running the Alpine version of the container image, and the check_ping command is not working.
It always results in the output: Could not interpret output from ping command

This issue can be solved by executing:
chmod u+s /bin/ping
chmod u+s /bin/ping6

Since the check_ping command is a popular official Nagios plugin, please consider adding this fix in the Dockerfile directly.

**Comment by manios** [comment]
Hi @BE-Digital-Solutions-Bart !

I have tried the latest image and it works fine in my case: `docker run -d -p 8080:80 --name nagios-amd64   manios/nagios:latest` . I get no error logs regarding the `check_ping` plugin. If you use the latest and you still see errors, then we need some more info:

## Image information

Can you please give us some information about the image you are using? In order to achieve this, please post here the output of the command:

```shell
docker inspect --format='{{range $k,$v := .Config.Labels}}{{printf "* %s: " $k}}{{if not $v}}{{$v}}{{else}}{{$v}}{{end}}{{"\n"}}{{end}}'  manios/nagios:[TAG]
```
Example output:

```
* build: 5
* gitCommit: c34e48cb6866b1a2bb77adf4d69feeab534ecc84
* homepage: https://www.nagios.com/
* maintainer: Christos Manios <maniopaido@gmail.com>
* nagiosPluginsVersion: 2.3.3
* nagiosVersion: 4.4.6
* name: Nagios
* nrpeVersion: 4.0.3
```

## Image architecture

What is the architecture your docker image is using? You can easily find out by executing this 

```shell
docker exec -it [container-name] /bin/sh -c 'apk --print-arch'
```
in your running container.

Example output:

```
amd64
```

## Relevant log output

Please copy and paste any relevant log output. This will be automatically formatted into code, so no need for markdown backticks. 
        
For example, in order to view the last 200 log lines from the container you can run:

```shell
docker logs --tail 200 [container-name]
```

You can also append the Apache error or access logs by issuing the commands:

```shell
docker exec -it [container-name] /bin/sh -c 'cat /var/log/apache2/error.log'

# AND

docker exec -it [container-name] /bin/sh -c 'cat /var/log/apache2/access.log
```

Thanks,
Chris

**Comment by BE-Digital-Solutions-Bart** [comment]
Hi Chris,


Thank you far the fast reply!

I haven't been able to reproduce the error this week. I have no idea what caused the issue exactly.

--- Details ---

I had been running the 4.5.1 version of your container image on my Synology NAS (model DS 920+) since over a year without any issues.
(And to be precies, I'm actually I was running a docker image created from a private fork of this project, including Python & nmap in the container image, and exposing web ui on a different port: bedigitalsolutionsbart/nagios:build-2024-03-23 on Docker Hub)

Last weekend, I had upgraded to the 4.5.9 version by merging in the latest changes, and encountered a few things that didn't work. (bedigitalsolutionsbart/nagios:build-2025-04-19 on Docker Hub)
I installed bash to fix 1 of my custom Nagios commands, and started receiving lots of Nagios notifications in MS Teams, e.g.

PROBLEM: telenet-router is DOWN
CRITICAL - Could not interpret output from ping command

generated by:

define command {
    command_name notify-host-by-teams
    command_line /usr/bin/python3 /opt/Custom-Nagios-Plugins/notify-teams/notify-teams-workflow.py "$_CONTACTWEBHOOKURL$" "$NOTIFICATIONTYPE$: $HOSTALIAS$ is $HOSTSTATE$" "$HOSTOUTPUT$"
}

When connecting to the container console, I could reproduce this error message by executing the "./libexec/check_ping" command.

I could fix this by executing: chmod u+s /bin/ping
This change immediately resulted in lots of RECOVERY notifications in Teams, indicating successful ping output, e.g.

RECOVERY: telenet-router is UP
PING OK - Packet loss = 0%, RTA = 1.88 ms - 192.168.0.1

---

I haven't been able to reproduce the error when running a fresh container on my NAS from image: bedigitalsolutionsbart/nagios:build-2025-04-19
I also haven't been able to reproduce the error when running a fresh container on my NAS of your container image: manios/nagios:build-30

When opening the terminal with "/bin/sh" and executing "apk --print-arch" results in:
x86_64


Thank you for the support, and feel free to close this issue.

Bart

**Comment by pygoubet** [comment]
Hi,
I am experiencing the same problem with image `manios/nagios:4.5.9` for linux/arm64 architecture. Within the container, as `nagios` user, the `ping` command fails. `chmod u+s /bin/ping` fixes it.

```bash
/opt/nagios # su -s /bin/sh - nagios -c "ping 192.168.1.1"
PING 192.168.1.1 (192.168.1.1): 56 data bytes
ping: permission denied (are you root?)

/opt/nagios # chmod u+s /bin/ping
/opt/nagios # su -s /bin/sh - nagios -c "ping 192.168.1.1"
PING 192.168.1.1 (192.168.1.1): 56 data bytes
64 bytes from 192.168.1.1: seq=0 ttl=62 time=1.041 ms
64 bytes from 192.168.1.1: seq=1 ttl=62 time=0.727 ms
```

As workaround, I have changed the container entrypoint command to: `sh -c `chmod u+s /bin/ping; /usr/local/bin/start_nagios'`.
